wdfn-server/waterdata/tests/test_views.py

```diff
@@ -142,6 +142,48 @@ def test_locations_list(self, client):
         assert text.count('01630500') == 16, 'Expected site 01630500 in output'
 
 
+class TestNetworkView(TestCase):
+    # pylint: disable=R0902
+
+    def setUp(self):
+        self.app_client = app.test_client()
+        self.test_url = '{0}/'.format(app.config['OGC_SERVICE'])
+        self.network = 'monitoring-locations'
+        self.format = 'json'
+
+    @mock.patch('waterdata.services.ogc.get_networks')
+    def test_networks(self, network_mock):
+        network_mock.return_value.status_code = 200
+        response = self.app_client.get('/networks/')
+        # Assert network listing works
+        assert response.status_code == 200
+
+    @mock.patch('waterdata.services.ogc.get_networks')
+    def test_one_network(self, network_mock):
+        network_mock.return_value.status_code = 200
+        response = self.app_client.get('/networks/{}/'.format(self.network))
+        # Assert a known network works in gui
+        assert response.status_code == 200
+
+    @mock.patch('waterdata.services.ogc.get_networks')
+    def test_invalid_network(self, network_mock):
+        network_mock.return_value.status_code = 200
+        network = 'monitoring-locations-invalid'
+        url = '/networks/{}/'.format(network)
+        response = self.app_client.get(url)
+        # Assert an invalid network in gui does not 404
+        assert response.status_code == 200
```

Comment:
So we are getting ahead of this JIRA as this part of the next but I think it should be a 404.